user-interface/src/main/java/life/qbic/datamanager/views/projects/project/measurements/registration/MeasurementUpload.java

```diff
@@ -258,54 +260,76 @@ private void runValidation(ArrayList<ValidationRequest> requests, int itemsToVal
-        .subscribe(responses -> displayValidationResults(
-            responses, itemsToValidate, fileName));
+        .doOnSuccess(it -> validationFinished(ui))
+        .subscribe(responses -> ui.access(() -> {
+          validationFinished(ui);
+          displayValidationResults(responses, itemsToValidate, fileName);
+          ui.push(); // Enforce server push
+        }));
   }
 
-  private void validationStarted() {
-    showValidationProgress();
-    setValidationProgressText("Starting validation ...");
+  private void validationStarted(UI ui) {
+    if (ui == null || !ui.isAttached()) {
+      return;
+    }
+    showValidationProgress(ui);
+    setValidationProgressText(ui, "Starting validation ...");
     isValidationInProgress.set(true);
   }
 
-  private void validationFinished() {
-    hideValidationProgress();
+  private void validationFinished(UI ui) {
+    if (ui == null || !ui.isAttached()) {
+      return;
+    }
+    hideValidationProgress(ui);
     isValidationInProgress.set(false);
   }
 
 
-  private void hideValidationProgress() {
-    getUI().ifPresent(ui -> ui.access(() -> validationProgress.setVisible(false)));
+  private void hideValidationProgress(@NonNull UI ui) {
+    if (ui.isAttached() && ui.getSession().hasLock()) {
+      validationProgress.setVisible(false);
+    } else {
+      ui.access(() -> validationProgress.setVisible(false));
+    }
   }
 
-  private void showValidationProgress() {
-    getUI().ifPresent(ui -> ui.access(() -> validationProgress.setVisible(true)));
+  private void showValidationProgress(@NonNull UI ui) {
+    if (ui.isAttached() && ui.getSession().hasLock()) {
+      validationProgress.setVisible(true);
+    } else {
+      ui.access(() -> validationProgress.setVisible(true));
+    }
   }
 
-  private void setValidationProgressText(String text) {
-    getUI().ifPresent(ui -> ui.access(() -> validationProgress.setText(text)));
+  private void setValidationProgressText(@NonNull UI ui, String text) {
+    if (ui.isAttached() && ui.getSession().hasLock()) {
+      validationProgress.setText(text);
+    } else {
+      ui.access(() -> {
+        validationProgress.setText(text);
+        ui.push();
+      });
+    }
   }
 
   private void displayValidationResults(List<ValidationResponse> responses, int totalValidations,
       String fileName) {
     ValidationResult result = ValidationResult.successful();
     var combinedResult = responses.stream().map(ValidationResponse::result)
         .reduce(result, ValidationResult::combine);
-
-    getUI().ifPresent(ui -> ui.access(() -> {
-      var measurementFileItem = new MeasurementFileItem(fileName,
-          new MeasurementValidationReport(totalValidations, combinedResult));
-      addAndDisplayFile(measurementFileItem);
-    }));
+    var measurementFileItem = new MeasurementFileItem(fileName,
+        new MeasurementValidationReport(totalValidations, combinedResult));
+    addAndDisplayFile(measurementFileItem);
```

Comment:
You removed the ui access here and introduced an ui.push in the calling code.
This is different from `validationFinished(UI ui)`. As these things always occur together wouldn't it make sense to handle them the same way?